Emi|GHG/CO2 subsector plots using CEDS UNFCCC data

pweigmann opened the pull request:
New compareScenario plots for 
- Emi|GHG|\<sector\>
- Emi|CO2|\<sector\>|\<subsector\>

... because CEDS2021 and UNFCCC now contains data to compare to.

Changed default: zeros are replaced by "NA" in historical data.

Increased compatibility with old _historical.mif_s by expanding conditional statements before plots.

fschreyer commented on R/compareScenarios.R:
```diff
@@ -1603,33 +1601,79 @@ compareScenarios <- function(mif, hist,
               "Emi|GHGtot|target|65% (Mt CO2-equiv/yr)"="-65% by 2030 (vs. 1990)")
+  
+  # remove as soon as renaming is done
+  if ("Emi|GHGtot (Mt CO2-equiv/yr)" %in% getNames(histData, dim=3)) {
```
Isn't this always true as some sources with ``Emi|GHGtot`` still hang around?

pweigmann replied:
True, I'll see if I can find a better solution.

fschreyer commented on R/compareScenarios.R:
```diff
@@ -1603,33 +1601,79 @@ compareScenarios <- function(mif, hist,
   swfigure(sw,print,p,sw_option="height=9,width=8")
 
+  ## ---- Emissions GHG Sectors ----
+  items <- c("Emi|GHG|Energy (Mt CO2eq/yr)",
```
Would be good to plot Emi|GHG|Land-Use Change here as well although there is no historic data for this yet to have all top-level IPCC sectors in this section.

pweigmann replied:
on it

fschreyer commented on R/compareScenarios.R:
```diff
@@ -1698,228 +1742,302 @@ compareScenarios <- function(mif, hist,
-                         ylab='Emi|CO2|Energy|Demand|Industry [Mt CO2/yr]',scales="free_y",plot.priority=c("x_hist","x","x_proj"),facet.ncol=3)
-  swfigure(sw,print,p,sw_option="height=9,width=8")
+    swlatex(sw,"\\subsubsection{Energy Supply}")
+    p <- mipLineHistorical(data[mainReg,,"Emi|CO2|Energy|Supply (Mt CO2/yr)"],x_hist=histData[mainReg,,"Emi|CO2|Energy|Supply (Mt CO2/yr)"][,,"historical.CEDS.Emi|CO2|Energy|Supply (Mt CO2/yr)"],
```
Why is the ``historical.CEDS`` necessary here? Wouldn't we want to plot from all sources which contain Emi|CO2|Energy|Supply?

pweigmann replied:
EDGAR, UNFCCC and UBA also have data here, but not for all regions, resulting in reduced GLO numbers. This was my (not very elegant) solution to only plot those sources that have data for all regions.
![image](https://user-images.githubusercontent.com/28896133/145796968-1a4a9838-8f20-40b3-acca-a37bd38d2fbf.png)

fbenke-pik replied:
FYI UNFCCC problem with zeros has been fixed in mrremind on friday and should no longer occur. i will take care of UBA soon as well.

fbenke-pik replied:
So I suggest fixing this in mrremind instead of here. Happy to provide an UBA fix asap.

pweigmann replied:
Sounds good, in that case I will remove the filter for certain sources.

fschreyer reviewed (changes requested):
Cool, looks good!

Was not entirely sure from taking a brief look. Are there plots doubled in the compare scenario PDF when CEDS/UNFCCC historical data is available, so once with and once without? That would actually be a bit confusing.

pweigmann commented:
> Are there plots doubled in the compare scenario PDF when CEDS/UNFCCC historical data is available, so once with and once without?

I don't think there is. In case there are doubled plots it's a mistake from my side.

pweigmann commented:
I found a better way to include both Emi|GHGtot and Emi|GHG in the plots, added a Emi|GHG|Land-Use Change plot and removed the filters that took care of regionally incomplete data as Falk will fix that on the mrremind side. I didn't find any doubled plots either, so I think we might be good to go?
